zebra/zebra_vty.c

```diff
@@ -1351,26 +1347,23 @@ DEFPY (show_interface_nexthop_group,
 
 DEFPY (show_nexthop_group,
        show_nexthop_group_cmd,
-       "show nexthop-group rib <(0-4294967295)$id|[<ip$v4|ipv6$v6>] [vrf <NAME$vrf_name|all$vrf_all>]>",
+       "show nexthop-group rib <(0-4294967295)$id [vrf <NAME$vrf_name|all$vrf_all>]>",
        SHOW_STR
        "Show Nexthop Groups\n"
        "RIB information\n"
        "Nexthop Group ID\n"
-       IP_STR
-       IP6_STR
        VRF_FULL_CMD_HELP_STR)
 {
 
 	struct zebra_vrf *zvrf = NULL;
-	afi_t afi = 0;
 
 	if (id)
 		return show_nexthop_group_id_cmd_helper(vty, id);
 
-	if (v4)
-		afi = AFI_IP;
-	else if (v6)
-		afi = AFI_IP6;
+	if (vrf_is_backend_netns() && (vrf_name || vrf_all)) {
+		vty_out(vty, "VRF subcommand does not make any sense in l3mdev based vrf's");
```

Comment:
this needs a terminating newline? and aren't errors usually preceded with `"%% ` - or is that not a convention?

Reply:
it's not a convention.

Reply:
isn't a newline ... reasonable?

the bigger question is the one about the treatment of singleton nhes - it does seem (to me) ok to want to see them by vrf. is there a specific problem introduced by offering that filter?